code-challenges.js

```diff
@@ -2,22 +2,50 @@
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
-var secretCodeWord1 = "lackadaisical"
+var secretCodeWord1 = "lakadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+//check each character and convert their value to a number if they are either a,e,i,o
 
-
+const convertCode = (str) => {
+ let splitString = str.split("");
+ 
+ let letters = splitString.map(val => { 
+     if (val.includes("a")) {
+         val = 4
+     } 
+     else if (val.includes("e")) {
+         val = 3
+     } 
+     else if(val.includes("i")) {
+         val = 1
+     } 
+     else if (val.includes("o")) {
+         val = 0
+     }
+     return val
+    })
+    return letters.join("")
+} 
+ 
+// console.log(convertCode(secretCodeWord1))
+// console.log(convertCode(secretCodeWord2))
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
+
+
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
 
+const letterA = (array) => {
+return array.map(val => val).filter(letter => letter.includes("a") || letter.includes("A"))
```

Comment:
in this one, you dont need a map _and_ a filter. you can just use filter. here's your code, with some minor adjustments:

```javascript
const letterA = (array) => {
  return array.filter(value => value.includes("a") || value.includes("A"))
}
console.log(letterA(arrayOfWords))
```